discretize/TreeMesh.py

```diff
@@ -632,8 +649,17 @@ def plotSlice(
             plt.show()
         return tuple(out)
 
-    def save(self, *args, **kwargs):
```

Comment:
Could this method stay and allow a user to pass a file name which will serialize any mesh to a file? For example:

```py
mesh = some_discretize_mesh_of_any_type
mesh.save('my_awesome_mesh.mesh')
```

Reply:
It was removed because It was originally overriding it here to explicitly say that save wouldn't work on the TreeMesh, now it behaves like the other meshes

Reply:
The `msh` and `mesh` extensions are greatly overused across viz libraries and other mesh analysis software. I think it would be best to step away from the `msh`/`mesh` convention I've seen used with discretize and towards something more distinct.

I think this PR would be a great place to standardize how meshes are saved - perhaps have the default extension to this `save` method be something like: `.dmesh`, `.discretize`, `.simpeg`

Reply:
> It was removed because It was originally overriding it here to explicitly say that save wouldn't work on the TreeMesh, now it behaves like the other meshes

Ah whoops - I wasn't looking closely enough  - I see now that `save` is indeed implemented here for all meshes

https://github.com/simpeg/discretize/blob/7c0af69f2c67ffe92e7759f93a9eb98df6d7fb85/discretize/base/base_mesh.py#L395

Reply:
My note about the file extension remains - instead of `.json`, would something more distinct be appropriate?

Reply:
Thanks @banesullivan: yes, the save / load should be working for all mesh-types at this point in discretize. 

I am inclined to leave the file extension as `json` as we can then take advantage of viewers and and tools for `json`. For example, in JupyterLab, there is a `json` viewer that nests properties nicely and such, but this by default will only work if the file extension is `json` (e.g. I renamed the one on the right to `.model`). 

![image](https://user-images.githubusercontent.com/6361812/62582825-96557700-b862-11e9-8f18-cf49266605d1.png)

Do you have a use-case in mind for why we would need it to be something more distinct?